src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List getAll(Class type) {
+        List<Bulldozer> bulldozers = new BulldozerProducer().get();
+        List<Excavator> excavators = new ExcavatorProducer().get();
+        List<Truck> trucks = new TruckProducer().get();
+        if (type.equals(Bulldozer.class)) {
+            return bulldozers;
+        }
+        if (type.equals(Excavator.class)) {
+            return excavators;
+        }
+        if (type.equals(Truck.class)) {
+            return trucks;
+        }
```

Comment:
We can use this  model - "else if"

Reply:
agree, at the moment you're always creating 3 lists of machines, let's use if-else to create and return a specific list